allensdk/brain_observatory/behavior/data_objects/licks.py

```diff
@@ -66,17 +81,23 @@ def from_stimulus_file(cls, stimulus_file: BehaviorStimulusFile,
         #
         # This bugfix copied from
         # https://github.com/AllenInstitute/visual_behavior_analysis/blob
-        # /master/visual_behavior/translator/foraging2/extract.py#L640-L647
+        # /master/visual_behavior/translator/foraging2/extract.py#L640-L647ß
+        if isinstance(stimulus_timestamps, np.ndarray):
+            lick_times = stimulus_timestamps
 
-        if len(lick_frames) > 0:
-            if lick_frames[-1] == len(stimulus_timestamps.value):
-                lick_frames = lick_frames[:-1]
-                cls._logger.error('removed last lick - '
-                                  'it fell outside of stimulus_timestamps '
-                                  'range')
+            max_length = min(len(lick_times), len(lick_frames))
 
-        lick_times = \
-            [stimulus_timestamps.value[frame] for frame in lick_frames]
+            lick_frames = lick_frames[0:max_length]
+            lick_times = lick_times[0:max_length]
+        else:
+            if len(lick_frames) > 0:
+                if lick_frames[-1] == len(stimulus_timestamps.value):
```

Comment:
It looks to me like this code is probably doing the same thing as the code above. This logic should not be depend on on if reading timestamps from `StimulusTimestamps` or `np.ndarray`. It would be nice if we could apply the same logic here when reading from `np.ndarray`.

Reply:
Not quite. The ndarray path just trims to the smallest of the two arrays. The other specifically checks to see if there is a final lick outside of the timestamps and removes it. This is important as the licks are loaded by frame the line below. Hopefully the need for this logic goes away once the StimulusTimestamps are fixed.

Reply:
Yes, but if the issue the if block is solving is to make them the same length and if lick_frames is only 1 greater than lick_times, then they both do the same thing. Can you confirm whether or not the `min` logic is needed to handle a different scenario?

Reply:
Likely yes and I can't really say this is needed for a different case. A consequence of inheriting Nathan's code. I'm hesitant to merge the logic partially because `len(stimulus_timestamps.value)` looks to be a longer array than `lick_frames` as the `lick_frames` end up being an index into stimulus_timestamps where as the input numpy array is just the lick times. We also don't check for if `lick_frames > stimulus_timestamps.value` but I suppose would be some very bad data.